hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/CostFunction.java

```diff
@@ -89,13 +89,14 @@ protected void regionMoved(int region, int oldServer, int newServer) {
    * @return The scaled value.
    */
   protected static double scale(double min, double max, double value) {
-    if (max <= min || value <= min) {
+    if (max <= min || value <= min
+      || Math.abs(max - min) <= 0.01 || Math.abs(value - min) <= 0.01) {
```

Comment:
Is 0.01 a suitable threshold? At least make a constant maybe called `DELTA`?

Reply:
values like `minCostNeedBalance` below are at 0.00 precision, so I think we should have an epsilon of at least 0.000 precision.

Reply:
Let me create a COST_EPSILON because I have seen quite wide range of precision.